src/gui/qgsqueryresultwidget.h

```diff
@@ -218,17 +225,151 @@ class GUI_EXPORT QgsQueryResultWidget : public QWidget, private Ui::QgsQueryResu
      */
     void updateButtons();
 
-    void showCellContextMenu( QPoint point );
-    void copySelection();
     void openQuery();
     void saveQuery( bool saveAs );
     void setHasChanged( bool hasChanged );
     void populatePresetQueryMenu();
+    void showHistoryPanel( bool show );
+
+  private:
+    QgsQueryResultPanelWidget *mQueryWidget = nullptr;
+    QPointer< QgsDatabaseQueryHistoryWidget > mHistoryWidget;
+
+    QMenu *mPresetQueryMenu = nullptr;
+
+    bool mHasChangedFileContents = false;
+
+    void updateDialogTitle();
+    void storeCurrentQuery( Qgis::QueryStorageBackend backend );
+
+    friend class TestQgsQueryResultWidget;
+};
+
+#ifndef SIP_RUN
+
+/**
+ * \ingroup gui
+ * \brief The QgsQueryResultWidget class allows users to enter and run an SQL query on a
+ * DB connection (an instance of QgsAbstractDatabaseProviderConnection).
+ *
+ * Query results are displayed in a table view.
+ * Query execution and result fetching can be interrupted by pressing the "Stop" push button.
+ *
+ * The widget supports a few QueryWidgetMode modes that pre-configure the widget appearance to
+ * be used in different contexts like when updating the SQL of an existing query layer.
+ *
+ * \note the ownership of the connection is transferred to the widget.
+ *
+ * \since QGIS 3.22
```

Comment:
Maybe it should be QGIS 3.44 as the new name of the class `QgsQueryResultPanelWidget` is different from the old class name `QgsQueryResultWidget`?

Reply:
Actually it's just a misleading diff -- QgsQueryResultWidget still exists and has the exact same API as before. It's just that the internals of it have been moved to the new QgsQueryResultPanelWidget class.